### PR TITLE
G0 5730 | Make customer name editable in edit invoice(sales/cash) flow

### PR DESCRIPTION
**Pull request by ravinderwalkover**
G0 5730 | Make customer name editable in edit invoice(sales/cash) flow

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 1]
```diff
@@ -1169,7 +1169,7 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
     }
 
     public openHeaderDropDown() {
-        if (!this.vm.selectedLedger.voucherGenerated) {
+        if (!this.vm.selectedLedger.voucherGenerated || this.vm.selectedLedger.voucherGeneratedType === "sales") {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:1172]: Consider using VoucherTypeEnum when comparing values

File: apps/web-giddh/src/app/store/Invoice/Receipt/receipt.reducer.ts [hunk 2]
```diff
@@ -134,6 +134,11 @@ export function Receiptreducer(state: ReceiptState = initialState, action: Custo
                             } else if (result.body.voucherDetails && m.voucherNumber === result.body.voucherDetails.voucherNumber) {
                                 m.grandTotal.amountForAccount = result.body.voucherDetails.grandTotal;
                             }
+
+                            if(result.body.type === VoucherTypeEnum.sales || result.body.type === VoucherTypeEnum.cash) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/store/Invoice/Receipt/receipt.reducer.ts:138]: Do add nullity check to result

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 3]
```diff
@@ -655,7 +655,7 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
 
         // get account details and set it to local var
         this.selectedAccountDetails$.subscribe(accountDetails => {
-            if (accountDetails && !this.isUpdateMode) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:658]: Please check it's impact

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts (hunk 1 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts]: Fixed

File: apps/web-giddh/src/app/store/Invoice/Receipt/receipt.reducer.ts (hunk 2 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/store/Invoice/Receipt/receipt.reducer.ts]: Fixed

**Pull request merged**